src/test/java/com/ultimatelogger/multiplatform/loggingifisonmultiprioritylogger/LoggingIfIsOnMultiPriorityLoggerTest.kt

```diff
@@ -1,24 +1,290 @@
 package com.ultimatelogger.multiplatform.loggingifisonmultiprioritylogger
 
 import com.nhaarman.mockitokotlin2.mock
+import com.ultimatelogger.multiplatform.UltimateLogger
+import com.ultimatelogger.multiplatform.di.LazyServiceLocator
+import com.ultimatelogger.multiplatform.output.LoggingIfIsOnMultiPriorityLogger
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.dsl.module
 import org.mockito.Mockito
 import org.mockito.verification.VerificationMode
-import com.ultimatelogger.multiplatform.output.LoggingIfIsOnMultiPriorityLogger
-import com.ultimatelogger.multiplatform.output.MultiPriorityLogger
 
-internal abstract class LoggingIfIsOnMultiPriorityLoggerTest(isLoggingOn: Boolean) {
+internal abstract class LoggingIfIsOnMultiPriorityLoggerTest(private var callCount: VerificationMode,
+                                                             private var shouldLog: Boolean) {
+
+    private val mockUltimateLogger: UltimateLogger = mock()
+
+    private val loggingIfIsOnMultiPriorityLogger = object : LoggingIfIsOnMultiPriorityLogger() {}
+
+    private val givenWithFileNameAndLineNum = true
+    private val givenWithClassName = true
+    private val givenWithMethodName = true
+    private val givenMsg = "[]];"
+    private val givenThrowable: Throwable = Exception()
+    private val givenAny = Any()
+
+    @Before
+    open fun setUp() {
+        loggingIfIsOnMultiPriorityLogger.shouldLog = shouldLog
+
+        val koinApplication = startKoin {
+            modules(
+                    module {
+                        single { mockUltimateLogger }
+                    }
+            )
+        }
+
+        LazyServiceLocator.koin = koinApplication.koin
+    }
+
+    @After
+    fun tearDown() = stopKoin()
+
+    @Test
+    fun `v(String) called callCount times v(String) on mockUltimateLogger`() {
```

Comment:
Sorry, my bad - grammatically correct will be `called v(String) callCount times`.